package.json

```diff
@@ -1,8 +1,13 @@
 {
   "dependencies": {
+    "@tanstack/react-query": "^5.59.20",
+    "@trpc/client": "11.0.0-rc.621",
+    "@trpc/react-query": "11.0.0-rc.621",
+    "@trpc/server": "11.0.0-rc.621",
     "prettier": "^3.3.3",
     "typescript": "^5.6.3",
-    "vite-node": "^2.1.4"
+    "vite-node": "^2.1.4",
+    "zod": "^3.23.8"
```

Comment:
we can use zod for type validation on frontend&backend. I added it here because it makes sense to have zod parse some types when they reach the server - in order to chekc that the type is what we expect!